frontends/open-discussions/package.json

```diff
@@ -0,0 +1,187 @@
+    "@babel/plugin-transform-react-jsx": "7.17.12",
+    "@babel/plugin-transform-runtime": "7.18.5",
+    "@babel/preset-env": "7.18.2",
+    "@babel/preset-flow": "7.17.12",
+    "@babel/preset-react": "7.17.12",
+    "@babel/register": "7.17.7",
+    "@babel/runtime": "7.18.3",
+    "@material/base": "0.2.3",
+    "@material/button": "0.3.11",
+    "@material/dialog": "0.27.0",
+    "@material/drawer": "0.36.0",
+    "@material/layout-grid": "0.34.0",
+    "@material/list": "0.2.14",
+    "@material/radio": "0.30.0",
+    "@material/rtl": "0.1.7",
+    "@material/snackbar": "0.44.0",
+    "@material/toolbar": "0.4.4",
+    "@mitodl/ckeditor-custom-build": "0.0.4",
+    "@mitodl/course-search-utils": "1.2.1",
+    "@mitodl/mdl-react-components": "0.0.3",
+    "@reduxjs/toolkit": "1.4.0",
+    "@sentry/browser": "5.5.0",
+    "@trust/webcrypto": "0.9.2",
+    "amplitudejs": "5.0.3",
+    "array-move": "2.2.0",
+    "assert": "2.0.0",
+    "autoprefixer": "9.7.4",
+    "babel-eslint": "10.1.0",
+    "babel-loader": "8.1.0",
+    "babel-plugin-istanbul": "4.1.6",
+    "blueimp-canvas-to-blob": "3.14.0",
+    "bodybuilder": "2.2.15",
+    "casual-browserify": "1.5.12",
+    "chai": "4.2.0",
+    "chai-as-promised": "7.1.1",
+    "cheerio": "1.0.0-rc.9",
+    "codecov": "3.7.1",
+    "core-js": "3.2.0",
+    "css-loader": "3.6.0",
+    "decimal.js-light": "2.5.0",
+    "detect-browser": "2.5.1",
+    "enzyme": "3.11.0",
+    "enzyme-adapter-react-16": "1.15.2",
+    "eslint": "7.5.0",
+    "eslint-config-google": "0.13.0",
+    "eslint-config-mitodl": "0.0.8",
+    "eslint-plugin-babel": "5.3.0",
+    "eslint-plugin-flow-vars": "0.5.0",
+    "eslint-plugin-flowtype": "4.2.0",
+    "eslint-plugin-mocha": "6.0.0",
+    "eslint-plugin-react": "7.14.3",
+    "eslint-plugin-react-hooks": "4.1.0",
+    "express": "4.16.3",
+    "fetch-mock": "6.4.4",
+    "flow-bin": "0.94.0",
+    "flow-typed": "2.4.0",
+    "focus-visible": "5.1.0",
+    "formik": "2.0.3",
+    "fuse.js": "6.0.0",
+    "history": "4.7.2",
+    "html-entities": "1.2.1",
+    "isomorphic-fetch": "2.2.1",
+    "jsdom": "16.6.0",
+    "jsdom-global": "3.0.2",
+    "locale-code": "2.0.1",
+    "lodash": "4.17.19",
+    "mini-css-extract-plugin": "0.5.0",
+    "mocha": "9.2.2",
+    "moment": "2.29.1",
+    "mutationobserver-shim": "0.3.2",
+    "nuka-carousel": "4.5.2",
+    "nyc": "15.1.0",
+    "object.entries": "1.0.4",
+    "os-browserify": "0.3.0",
+    "path-browserify": "1.0.1",
+    "postcss-loader": "2.1.5",
+    "prettier-eslint-cli": "5.0.0",
+    "prop-types": "15.6.2",
+    "prosemirror-commands": "^1.0.7 <=1.1.7",
+    "prosemirror-example-setup": "^1.0.1 <= 1.1.2",
+    "prosemirror-markdown": "^1.2.2 <=1.5.1",
+    "prosemirror-model": "^1.6.6 <=1.14.1",
+    "prosemirror-schema-basic": "^1.0.0 <=1.1.2",
+    "prosemirror-schema-list": "^1.0.1 <=1.1.4",
+    "prosemirror-state": "^1.2.1 <=1.3.4",
+    "prosemirror-view": "^1.3.7 <=1.18.7",
```

Comment:
I had a lot of trouble with package versions in this PR.

When I upgraded from yarn 1 to yarn 3, I rebuilt the lockfile and a lot of things got minor version upgrades. Possibly I could have updated the lockfile with fewer changes.

Subsequently I had some issues with building and tests. Switching to specific versions (not `^`) fixed most of them—presumably getting the lockfile closer to what it was originally. But specific versions caused problems for prosemirror. The prosemirror packages have `prosemirror-state` as a dependency, and pegging specific versions in our package.json meant that different prosemirror packages were using different versions/instances of `prosemirror-state`, which led to some errors.

Using version ranges for prosemirror means they all use the same version of prosemirror-state.

I capped the ranges because I was getting import & type errors with too high a minor version.